### DISCUSSION TITLE
Sniffing 是否可以考虑增加“仅限特定域名重置目标地址”的功能

### DISCUSSION THREAD

**Discussion by Hyyyy11** [XTLS/Xray-core#2746] [Ideas]
部分机场所使用的流媒体 DNS 解锁方案必须要开启 Sniffing，且要关闭 routeOnly（比如我正在用的某 N 开头超大型机场🤡）。开启 routeOnly 后无法解锁，但我想使用 routeOnly 这个功能，所以是否可以考虑增加一个类似 domainsExcluded 的选项，在开启 routeOnly 后，这个选项中的域名仍然会重置目标地址。

**Comment by chise0713** [comment]
你可以考虑使用不同的inbound来做到你需要的效果，本地使用switchyomega分流，或者在xray内套娃分流。
<details><summary>一个xray分流的例子</summary>

inbounds:
```jsonc    
    {
      "tag": "socks-in",
      "port": 10808,
      "protocol": "socks",
      "settings": {
        "auth": "noauth",
        "udp": true,
        "allowTransparent": false
      },
      "streamSettings": {
        "sockopt": {
          "tcpFastOpen": true
        }
      }
    },
        {
                "listen": "127.0.0.1",
                "port": 853,
                "protocol": "dokodemo-door",
                "settings": {
                        "network": "tcp"
                        "port": 443,
                },
                "tag": "netflix-in"
      "sniffing": {
        "enabled": true,
        "destOverride": ["http", "tls", "quic"],
        "routeOnly": false
    },
        }

```
outbounds:
```jsonc
    {
      "protocol": "freedom",
      "settings": {
     "redirect": "127.0.0.1:853",
    }
      "tag": "netflix"
    }
```
routing:
```jsonc
      {
        "type": "field",
        "inboundTag": ["socks"],
        "domain": "netflix.com"
        "outboundTag": "netflix"
      },
      {
        "type": "field",
        "inboundTag": ["netflix-in"],
        "outboundTag": "proxy"
      },

```
</details>

**Comment by us254** [comment]
Currently, Xray does not support domain name matching in the routing sense with the `domainsExcluded` option. 

However, you can achieve your desired functionality by using the combination of `routeOnly` and `destOverride` options. By enabling `routeOnly` and `destOverride`, and setting the `domainStrategy` to `AsIs`, you can ensure that the sniffed domain name is used for routing only, while keeping the target address as the IP address.

when sniffing is enabled, Xray will sniff the domain name from the traffic data and reset the target address accordingly. To prevent this reset for specific domain names, you can consider adding those domain names to a list and excluding them from being reset using a custom implementation.
